model/task/task.go

```diff
@@ -2651,44 +2651,147 @@ func (t *Task) MarkUnscheduled() error {
 
 }
 
-// MarkAllForUnattainableDependency updates many tasks (taskIDs) to mark
-// the dependency (dependencyID) as attainable or not. If marking a dependency
-// unattainable, it creates an event log for each newly blocked task. This
-// returns all the tasks after the update.
-func MarkAllForUnattainableDependency(ctx context.Context, tasks []Task, dependencyID string, unattainable bool) ([]Task, error) {
+// MarkAllForUnattainableDependency updates many tasks (taskIDs) to mark a
+// subset of all their dependencies (dependencyIDs) as attainable or not. If
+// marking the dependencies unattainable, it creates an event log for each newly
+// blocked task. This returns all the tasks after the update.
+func MarkAllForUnattainableDependencies(ctx context.Context, tasks []Task, dependencyIDs []string, unattainable bool) ([]Task, error) {
 	if len(tasks) == 0 {
 		return tasks, nil
 	}
 
+	dependencyIDSet := map[string]struct{}{}
+	for _, depID := range dependencyIDs {
+		dependencyIDSet[depID] = struct{}{}
+	}
+
+	toUpdate := getTaskDependenciesToUpdate(tasks, dependencyIDSet, unattainable)
+
+	if err := updateTaskDependenciesChunked(ctx, toUpdate, unattainable); err != nil {
+		return nil, errors.Wrap(err, "updating task dependencies in chunks")
+	}
+
+	event.LogManyTasksBlocked(ctx, toUpdate.newlyBlockedTaskData)
+
+	updatedTasks, err := FindAll(db.Query(ByIds(toUpdate.taskIDs)))
+	if err != nil {
+		return nil, errors.Wrap(err, "finding updated tasks")
+	}
+	if len(updatedTasks) != len(tasks) {
+		return nil, errors.Errorf("updated dependencies for tasks but subsequent query for updated tasks returned a different number of tasks (%d) than expected (%d)", len(updatedTasks), len(tasks))
```

Comment:
Should we log this error instead of returning it? If this is likely to occur because of a logic bug, this could prevent us from continuing to helpful operations, like updating build and version status

Reply:
Makes sense! Changed it to an error log.